src/Vault.sol

```diff
@@ -431,8 +431,8 @@ contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, Reen
         view
         returns (uint256, uint256)
     {
-        uint256 baseAssets = _convertAssetToBase(asset_, shares);
-        uint256 assets = baseAssets.mulDiv(totalAssets() + 1, totalSupply() + 10 ** 0, rounding);
+        uint256 assets = shares.mulDiv(totalAssets() + 1, totalSupply() + 10 ** 0, rounding);
+        uint256 baseAssets = _convertAssetToBase(asset_, assets);
```

Comment:
@danoctavian another approach that might be better is to add the reverse of _convertAssetToBase....

_convertBaseToAsset(_asset, shares);

It seems like converting the assets here, which might be any given _asset, is wrong.

Reply:
Yes that is correct we need `_convertAssetToBase`

Reply:
One issue here why this wasn't caught i believe is because we don't have fuzz testing for  this

I'd consider making a public `    function _convertToAssets(address asset_, uint256 shares, Math.Rounding rounding)` function and fuzz test that

Or a version of function mint, that also specifies the asset, to be equivalent to depositsset

Reply:
There is fuzz testing for this, but I think it needs to be tested via scenario based testing that simulates accounting flows.

https://github.com/yieldnest/yieldnest-vault/blob/cfdec9e0c6a30a08e5d1c34ae071374f0bc5606c/test/mainnet/invariants.spec.sol#L202

Reply:
The converToAssets hardcodes asset_ to asset(), which is base asset hence why it doesn't test the whole functionality of `function _convertToAssets(address asset_, uint256 shares, Math.Rounding rounding)`

Reply:
Will open a new PR to add mintAsset and previewMintAsset